Kernel/Devices/EBRPartitionTable.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2020, Liav A. <liavalb@hotmail.co.il>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <AK/RefPtr.h>
+#include <AK/Vector.h>
+#include <Kernel/Devices/DiskDevice.h>
+#include <Kernel/Devices/DiskPartition.h>
+#include <Kernel/Devices/MBRPartitionTable.h>
+
+struct [[gnu::packed]] EBRPartitionExtension
+{
+    u8 unused_area[446];
+    MBRPartitionEntry entry;
+    MBRPartitionEntry next_chained_ebr_extension;
+    MBRPartitionEntry unused[2];
+    u16 mbr_signature;
+};
+
+class EBRPartitionTable {
+
+public:
+    EBRPartitionTable(NonnullRefPtr<DiskDevice>);
+    ~EBRPartitionTable();
+
+    bool initialize();
+    RefPtr<DiskPartition> partition(unsigned index);
+
+private:
+    int get_ebr_container() const;
```

Comment:
I would call this `index_of_ebr_container` or `find_index_of_ebr_container` since what it really returns is an index, not an EBR container.

Reply:
I'd call this `get_index_of_ebr_container` as a compromise :)

Reply:
Please avoid using the `get_` prefix in getters. https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#right-3

Reply:
OK, so `index_of_ebr_container` is probably the best choice :)